### PR TITLE
task 5 complete

### PR DESCRIPTION
**Pull request by Asbayazitog1**

### CODE REVIEW

File: components/single-article-main.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from "react"
+import { getArticlesByArticleID } from "../api"
+
+const SingleArticleMain =({article_id})=>{
+    const[ isloading,setIsloading] =useState(true)
+    const [article,setArticle]=useState({})
+   useEffect(()=>{
+    console.log(article_id)
```
Comment (popatre) [inline components/single-article-main.jsx]: remove log

**Review by popatre — CHANGES_REQUESTED** [review]
- [ ] We need to get the article id from the url. At the moment, when i refresh the page, the article does not load correctly. How would i currently send someone a certain article?
- [ ] We need to fetch the article id from the api, and not from state. Could have stale data via state -

https://notes.northcoders.com/courses/js-front-end/react-router#parametric-endpoints-useparams

File: components/article-list.jsx [hunk 2]
```diff
@@ -16,16 +16,14 @@ const ArticleList =({setArticle_id}) =>{
...
+    
 return <ul id="article-list">
             {    
             articles.map(article =>{
                 return <section key={article.article_id} id="artile-list-item">
-                    <Link to={{pathname:'/single-article-main', }}>
-                    <li id="" onClick={()=>{handleClick(article.article_id)}}>{article.title}</li>
-                    <img src={article.article_img_url} alt="" id="article-img" onClick={()=>{handleClick(article.article_id)}} />
+                    <Link to={`/articles/${article.article_id}`}>
+                    <li id="">{article.title}</li>
```
Comment (ghost) [inline components/article-list.jsx:25]: maybe don't leave ids with empty strings.

**Review by ghost — APPROVED** [review]
Nice fix, go ahead and merge.

**Pull request merged**